### DISCUSSION TITLE
Exposing library functionality as a hook (can no longer use useObserver)

### DISCUSSION THREAD

**Discussion by jezzgoodwin** [mobxjs/mobx#2887] [Q&A] [answered]
Hi all,

I use MobX to write various features for my company app, but not everyone here necessarily uses MobX. Previously I have created custom hooks which expose some of my library's features for use by other devs and under the hood I have used `useObserver` to watch for changes.

In the latest version of MobX this function is now deprecated. I ideally want to avoid having to ask other devs to wrap their components in `observer` as it feels like it's leaking implementation details.

Is there a way for me to access useObserver without getting the deprectated message maybe?

Cheers
Jezz

**Comment by urugator** [comment]
Please see the discussion here https://github.com/mobxjs/mobx/discussions/2566#discussioncomment-557672

**Reply by jezzgoodwin** [reply]
I have followed this up with a comment in the other discussion. See here https://github.com/mobxjs/mobx/discussions/2566#discussioncomment-593576

**Comment by jezzgoodwin** [comment, marked as answer]
This has been answered here https://github.com/mobxjs/mobx/discussions/2566#discussioncomment-593670 with an example utility hook.
